File: shared/types/income.ts
export interface Income {
  id: string;
  userId: string;
  monthId: string;
  source: string;
  amount: number;
  incomeDate: string;
  comment: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateIncomeRequest {
  source: string;
  amount: number;
  incomeDate: string;
  comment?: string;
}

export interface UpdateIncomeRequest {
  source?: string;
  amount?: number;
  incomeDate?: string;
  comment?: string;
}

File: backend/src/HowAreMyFinances.Api/Models/Income.cs
namespace HowAreMyFinances.Api.Models;

public sealed record Income(
    Guid Id,
    Guid UserId,
    Guid MonthId,
    string Source,
    decimal Amount,
    DateOnly IncomeDate,
    string? Comment,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record CreateIncomeRequest(
    string Source,
    decimal Amount,
    DateOnly IncomeDate,
    string? Comment
);

public sealed record UpdateIncomeRequest(
    string? Source,
    decimal? Amount,
    DateOnly? IncomeDate,
    string? Comment
);
